src/openrct2-ui/input/MouseInput.cpp

```diff
@@ -1253,67 +1253,67 @@ void input_state_widget_pressed(
                     auto wClass = w->classification;
                     auto wNumber = w->number;
                     int32_t dropdown_index = 0;
+                    bool dropdownCleanup = false;
 
                     if (w->classification == WC_DROPDOWN)
                     {
                         dropdown_index = dropdown_index_from_point(screenCoords, w);
-                        if (dropdown_index == -1)
-                        {
-                            goto dropdown_cleanup;
-                        }
-
-                        if (dropdown_index < DROPDOWN_ITEMS_MAX_SIZE && dropdown_is_disabled(dropdown_index))
-                        {
-                            goto dropdown_cleanup;
-                        }
-
-                        if (gDropdownItemsFormat[dropdown_index] == DROPDOWN_SEPARATOR)
-                        {
-                            goto dropdown_cleanup;
-                        }
+                        dropdownCleanup = dropdown_index == -1
+                            || (dropdown_index < DROPDOWN_ITEMS_MAX_SIZE && dropdown_is_disabled(dropdown_index))
+                            || gDropdownItemsFormat[dropdown_index] == DROPDOWN_SEPARATOR;
                     }
                     else
                     {
                         if (cursor_w_class != w->classification || cursor_w_number != w->number
                             || widgetIndex != cursor_widgetIndex)
-                            goto dropdown_cleanup;
-                        dropdown_index = -1;
-                        if (_inputFlags & INPUT_FLAG_DROPDOWN_STAY_OPEN)
                         {
-                            if (!(_inputFlags & INPUT_FLAG_DROPDOWN_MOUSE_UP))
+                            dropdownCleanup = true;
+                        }
+                        else
```

Comment:
This wasn't an else before, is this correct?

Reply:
The diff here didn't become the most easy to see, but it kind of was, if you take a close look at the code:
https://github.com/OpenRCT2/OpenRCT2/blob/01d0287ee2a135e8f337c75df4834f92da23093c/src/openrct2-ui/input/MouseInput.cpp#L1277-L1289

If it matched the `if`, it went to the `goto`, which just skipped everything below, otherwise it did the `else` logic. So it does look like an `if...else` to me